cdc/log.cc

```diff
@@ -538,23 +563,136 @@ class transformer final {
                                 if (view.is_live_and_has_ttl()) {
                                     ttl = view.ttl();
                                 }
-                            } else {
-                                op = column_op::del;
-                            }
-
-                            values[0] = data_type_for<column_op_native_type>()->decompose(data_value(static_cast<column_op_native_type>(op)));
-                            res.set_cell(log_ck, *dst, atomic_cell::make_live(*dst->type, ts, tuple_type_impl::build_value(values)));
-
-                            if (pirow && pirow->has(cdef.name_as_text())) {
-                                values[0] = data_type_for<column_op_native_type>()->decompose(data_value(static_cast<column_op_native_type>(column_op::set)));
-                                values[1] = pirow->get_blob(cdef.name_as_text());
-
-                                assert(std::addressof(res.partition().clustered_row(*_log_schema, *pikey)) != std::addressof(res.partition().clustered_row(*_log_schema, log_ck)));
-                                assert(pikey->explode() != log_ck.explode());
-                                res.set_cell(*pikey, *dst, atomic_cell::make_live(*dst->type, ts, tuple_type_impl::build_value(values)));
-                            }
+                            } 
                         } else {
-                            cdc_log.warn("Non-atomic cell ignored {}.{}:{}", _schema->ks_name(), _schema->cf_name(), cdef.name_as_text());
+                            auto mv = cell.as_collection_mutation();
+                            auto now = gc_clock::time_point::min();
+                            std::optional<column_op> oop;
+                            op = column_op::add;
+
+                            std::vector<bytes> buf;
+                            values[1] = mv.with_deserialized(*cdef.type, [&](collection_mutation_view_description view) -> bytes_opt {
+                                if (view.tomb) {
+                                    // there is a tombstone with timestamp before this mutation.
+                                    // this is how a assign collection = <value> is represented.
+                                    oop = column_op::set;
+                                }
+                                auto process = [&](auto value_callback) {
```

Comment:
`process` is not a really descriptive name

Reply:
The surrounding function is called "transform". I think the descriptiveness is clear enough given the surrounding scope.